### DISCUSSION TITLE
ESP32 crashes with NIMBLE scan when more than some 400 devices nearby

### DISCUSSION THREAD

**Discussion by joopheuvel** [h2zero/NimBLE-Arduino#593] [Q&A]
Have been using ESP32 (Olimex ESP32POE) succesfully to scan BLE devices nearby.
I am using pretty much the standard example core code for this with the callback, init and scan.
Devices is connected to Ethernet and am using scan window that is maximized 100% (90 or 99 also has same issue, am not using WiFi so do not need to switch antenna), interval 200ms (tried different settings here too, and a scantime varying from 1sec to 5, to 10 sec.
Obviously for 5-10 sec I get many results that I store in an effective array (no slow processing in the loop), and do the processing of
the results when the full scan is done. The heap is 50k (have firebase and blynk running too, fragmentation unsure but should be okay). It works fine up until that number of 400 than it is simply too much for the system. This is for a festival entrance and indeed (as also in the main stages, there are many many devices (phones) around.
Any idea what can be done in this situation?
Thanks.
Joop

**Comment by joopheuvel** [comment]
Does anybody know if the issue that was in the ble code was also fixed in the nimBLE code? See "ble crashes in dense environments" saknarak/arduino-esp32@38461c0

**Reply by h2zero** [reply]
Do you have any logs of this crash?

**Comment by joopheuvel** [comment]
Unfortunately not as the devices were on site, remote, when they crashed, no serial dump. Fir the moment I have disabled active scanning and playing around with scan interval and window and ensuring heap is >100k. Can it be that the callbacks come to quick and system is not able to cope with amount of data/callbacks? How many devices can be tracked in a scan, any idea? Thanks. I can let you know after weekend if approach above changed anything.

**Reply by h2zero** [reply]
Without having a coredump to reference I can only speculate that this is a heap issue somewhere or possibly watchdog. It could also be a a BLE controller fault but we have no way to know. Wish I could be more help but I would need more data.

**Comment by joopheuvel** [comment]
Thanks for following up. I have done a test with a heap of 100k and it was
okay. It was in a less dense environment but got no reboots. I also set
activescan off to try to reduce effort from esp32 side. The discussion can
be closed. Thanks for now!
Jool
